skiko/src/awtMain/cpp/windows/exceptions_handler.cc

```diff
@@ -47,19 +43,35 @@ const char *getDescription(DWORD code)
         return "EXCEPTION_SINGLE_STEP";
     case EXCEPTION_STACK_OVERFLOW:
         return "EXCEPTION_STACK_OVERFLOW";
+    case ERROR_MOD_NOT_FOUND:
+        return "ERROR_MOD_NOT_FOUND";
     default:
         return "UNKNOWN EXCEPTION";
     }
 }
 
-void throwJavaException(JNIEnv *env, const char *function, DWORD sehCode)
-{
-    char buffer[200];
-    int result = snprintf(
-        buffer, sizeof(buffer) - 1, "Native exception in [%s]:\nSEH description: %s\n", function, getDescription(sehCode));
-    static jclass logClass = (jclass) env->NewGlobalRef(env->FindClass("org/jetbrains/skiko/RenderExceptionsHandler"));
-    static jmethodID logMethod = env->GetStaticMethodID(logClass, "throwException", "(Ljava/lang/String;)V");
-    env->CallStaticVoidMethod(logClass, logMethod, env->NewStringUTF(buffer));
+void throwJavaRenderExceptionByExceptionCode(JNIEnv *env, const char *function, DWORD code) {
+    char fullMsg[200];
+    int result = snprintf(fullMsg, sizeof(fullMsg) - 1,
+        "Native exception in [%s], code %lu: %s", function, code, getDescription(code));
+
+    static jclass cls = (jclass) env->NewGlobalRef(env->FindClass("org/jetbrains/skiko/RenderExceptionsHandler"));
+    static jmethodID method = env->GetStaticMethodID(cls, "throwException", "(Ljava/lang/String;)V");
+    env->CallStaticVoidMethod(cls, method, env->NewStringUTF(fullMsg));
 }
 
-#endif
\ No newline at end of file
+
+void throwJavaRenderExceptionByErrorCode(JNIEnv *env, const char *function, DWORD code) {
+    char fullMsg[1024];
+    char *msg = 0;
+    FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
+        NULL, code, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR) &msg, 0, NULL);
```

Comment:
```suggestion
        NULL, code, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR) &msg, 1024, NULL);
```
You should pass the size to avoid stack overflow

Reply:
Or even like that. `sizeof` should work for this case
```suggestion
        NULL, code, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR) &msg, sizeof(msg), NULL);
```

Reply:
> sizeof(msg),

msg isn't allocated, we can't use it.

```
You should pass the size to avoid stack overflow
```
Why there will be a stack overflow?

[The doc](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage) says that it represents the minimal allocated size of a string:
```
If FORMAT_MESSAGE_ALLOCATE_BUFFER is set, this parameter specifies the minimum number of TCHARs to allocate for an output buffer.
```

Reply:
> msg isn't allocated

it is.

> Why there will be a stack overflow?

*buffer overflow

Reply:
Ah, `FORMAT_MESSAGE_ALLOCATE_BUFFER` flag. then you don't need to allocate the buffer in stack

Reply:
nvm, it was `fullMsg`